suds/transport/sslverify.py

```diff
@@ -0,0 +1,168 @@
+import logging
+import re
+
+from OpenSSL import SSL
+
+from twisted.python.failure import Failure
+
+from twisted.internet.ssl        import Certificate
+from twisted.internet._sslverify import (ClientTLSOptions, OpenSSLCertificateOptions,
+                                         _maybeSetHostNameIndication, SSL_CB_HANDSHAKE_START,
+                                         SSL_CB_HANDSHAKE_DONE)
+
+
+log = logging.getLogger(__name__)
+
+
+def optionsForClientTLS(hostname, trustRoot=None, clientCertificate=None, **kw):
+    """
+    Reimplemented from twisted.internet.ssl to allow extra parameters to be passed correctly.
+
+    @return: A client connection creator.
+    @rtype: L{IOpenSSLClientConnectionCreator}
+    """
+    extraCertificateOptions = kw.pop('extraCertificateOptions', None) or {}
+    if kw:
+        raise TypeError(
+            "optionsForClientTLS() got an unexpected keyword argument"
+            " '{arg}'".format(
+                arg=kw.popitem()[0]
+            )
+        )
+    if not isinstance(hostname, unicode):
+        raise TypeError(
+            "optionsForClientTLS requires text for host names, not "
+            + hostname.__class__.__name__
+        )
+    if clientCertificate:
+        extraCertificateOptions.update(
+            privateKey=clientCertificate.privateKey.original,
+            certificate=clientCertificate.original
+        )
+
+    # Only pass the trustRoot if it is not None to avoid mutually exclusive param issues.
```

Comment:
The default code passes a `trustRoot` param to `OpenSSLCertificateOptions` regardless of whether it is None or not.

The twisted code makes use of some decorators on the initializer that look at the kwargs passed in, and will raise an exception if certain pairs of arguments are passed.  Merely having `trustRoot` in the kwargs, even though the value is `None`, will cause problems when passing `verify`, `caCerts`, or `requireCertificate`.